presto-iceberg/src/main/java/io/prestosql/plugin/iceberg/IcebergTransactionHandle.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.iceberg;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import io.prestosql.spi.connector.ConnectorTableHandle;
+import io.prestosql.spi.connector.ConnectorTransactionHandle;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import java.util.UUID;
+
+import static java.util.Objects.requireNonNull;
+
+public class IcebergTransactionHandle
+        implements ConnectorTransactionHandle
+{
+    private final UUID uuid;
+    private Map<ConnectorTableHandle, Long> tableSnapshotIdMap; // Used for refresh of a materialized view
```

Comment:
The purpose of the transaction handle is so that the engine can associate the `beginTransaction()`, `getMetadata()` and `commit()` calls. However, it's just a handle and shouldn't the store.

Instead, it's expected that the transaction state is stored inside `IcebergMetadata`. That's why `IcebergConnector` creates the metadata object once and stores it in a map using the handle.

Reply:
Correct. We discussed this offline as well. Removed all changes to stored metadata on transaction handle. Moved the metadata to IcebergMetadata.